core/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```diff
@@ -59,35 +156,47 @@ public RegexpQueryBuilder flags(RegexpFlag... flags) {
                 value |= flag.value;
             }
         }
-        this.flags = value;
+        this.flagsValue = value;
+        return this;
+    }
+
+    public RegexpQueryBuilder flags(int flags) {
+        this.flagsValue = flags;
         return this;
     }
 
+    public int flags() {
+        return this.flagsValue;
+    }
+
     /**
      * Sets the regexp maxDeterminizedStates.
      */
     public RegexpQueryBuilder maxDeterminizedStates(int value) {
         this.maxDeterminizedStates = value;
-        this.maxDetermizedStatesSet = true;
```

Comment:
I think you can answer for yourself at least for this one: would you be comfortable with 2.0 released with still this bug in it? Mayeb this fix should even be backported to 1.x, it's a bug and should be treated as such